### DISCUSSION TITLE
Color by vector components

### DISCUSSION THREAD

**Discussion by WubbzyFromWuzzleburg** [Kitware/trame#305] [Trame/VTK] [answered]
Hi, I'm working on a project similar to 04_application in the Trame tutorial. The "Color by" VSelect box seems to color the vtk image in the window by the scalars in the vtk file. Besides scalars, there are vectors in my vtk file. Can I color the image using the x, y, z components of the vectors? That is, for each vector, I want to add three items to the "Color by" VSelect box. For example, if there's a vector named "u_v_w", there should be "u_v_x-X" for the x component, "u_v_x-Y" for the Y component, and "u_v_x-Z" for the z component in the "Color by" box. Thanks!

**Comment by jourdain** [comment, marked as answer]
trame is just putting a UI to your VTK/Python so it is up to you to create and bind the UI in a way that make sense to your use case. (Maybe for vectors, you want to add a drop down with (magnitude, x, y, z). 

If you are looking for a VTK examples, you can [browse them here](https://examples.vtk.org/site/).
